experanto/datasets.py

```diff
@@ -438,11 +447,219 @@ def __getitem__(self, idx) -> dict:
+
+        self._screen_sample_times = np.arange(
+            self.start_time, self.end_time, 1.0 / self.sampling_rates["screen"]
+        )
+
+        self.transforms = self.initialize_transforms()
+
+        # Dynamically filter swap times based on meta_conditions
+        self._swap_times_valid_trial = self.get_swap_time_valid_conditions()
+        self.DataPoint = namedtuple("DataPoint", self.device_names)
+
+    def _read_trials(self) -> None:
+        screen = self._experiment.devices["screen"]
+        self._trials = [t for t in screen.trials]
+        start_idx = np.array([t.first_frame_idx for t in self._trials])
+        self._start_times = screen.timestamps[start_idx]
+        self._end_times = np.append(screen.timestamps[start_idx[1:]], np.inf)
+        self.meta_conditions = {}
+        for k in self.modality_config.screen.valid_condition.keys():
+            self.meta_conditions[k] = [t.get_meta(k) if t.get_meta(k) is not None else "blank" for t in self._trials]
+
+    def initialize_statistics(self) -> None:
+        self._statistics = {}
+        for device_name in self.device_names:
+            self._statistics[device_name] = {}
+            if self.modality_config[device_name].transforms.get("normalization", False):
+                mode = self.modality_config[device_name].transforms.normalization
+                assert mode in ['standardize', 'normalize']
+                means = np.load(self._experiment.devices[device_name].root_folder / "meta/means.npy")
+                stds = np.load(self._experiment.devices[device_name].root_folder / "meta/stds.npy")
+                if mode == 'standardize':
+                    means = np.zeros_like(means)
+
+                self._statistics[device_name]["mean"] = means.reshape(1, -1)
+                self._statistics[device_name]["std"] = stds.reshape(1, -1)
 
+    def initialize_transforms(self):
+        transforms = {}
+        for device_name in self.device_names:
+            if device_name == "screen":
+                add_channel = Lambda(lambda x: x[:, None, ...] if isinstance(x, torch.Tensor) and len(x.shape) == 3
+                                 else torch.from_numpy(x[:, None, ...]) if isinstance(x, np.ndarray) and len(x.shape) == 3
+                                 else x)
+                transform_list = [v for v in self.modality_config.screen.transforms.values() if isinstance(v, torch.nn.Module)]
+                transform_list.insert(0, add_channel)
+                transforms[device_name] = Compose(transform_list)
+            else:
+                transform_list = [ToTensor()]
+            if self.modality_config[device_name].transforms.get("normalization", False):
+                transform_list.append(
+                    torchvision.transforms.Normalize(self._statistics[device_name]["mean"], self._statistics[device_name]["std"])
+                )
+            transforms[device_name] = Compose(transform_list)
+        return transforms
```

Comment:
Same here, these seems to be a copypaste from [lines 372-393](https://github.com/sensorium-competition/experanto/blob/33b3c00ce2307dbeef6b63b33596c1e89512e14b/experanto/datasets.py#L372-L393) from the ChunkDataset. I would suggest to move it to a parent class and reuse